pkg/cmd/provisioning/alibabacloud/attach-ram-policy.go

```diff
@@ -0,0 +1,293 @@
+	componentSk := os.Getenv(componentAccessKeySecret)
+	if componentAk == "" || componentSk == "" {
+		log.Fatalf("Empty value of CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_ID or CCOCTL_ALIBABA_COMPONENT_ACCESS_KEY_SECRET")
+	}
+	AttachRAMPolicyOpts.ComponentAccessKeyId = componentAk
+	AttachRAMPolicyOpts.ComponentAccessKeySecret = componentSk
+	//call sts GetCallerIdentity for the username of given component ak/sk
+	stsClient, err := sts.NewClientWithAccessKey(AttachRAMPolicyOpts.Region, componentAk, componentSk)
+	if err != nil {
+		log.Fatalf("Failed to init sts client from ram username: %++v", refineMissingRegionIdErr(err))
+	}
+	req := sts.CreateGetCallerIdentityRequest()
+	req.Scheme = apiRequestScheme
+	res, err := stsClient.GetCallerIdentity(req)
+	if err != nil {
+		log.Fatalf("Failed to get caller identity: %++v", err)
+	}
+	userName, err := getRAMUserName(res.Arn)
+	if err != nil {
+		log.Fatalf("Failed to get ram user name: %++v", err)
+	}
+	AttachRAMPolicyOpts.UserName = userName
+	if AttachRAMPolicyOpts.TargetDir == "" {
+		pwd, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Failed to get current directory: %s", err)
+		}
+
+		AttachRAMPolicyOpts.TargetDir = pwd
+	}
+
+	fPath, err := filepath.Abs(AttachRAMPolicyOpts.TargetDir)
+	if err != nil {
+		log.Fatalf("Failed to resolve full path: %s", err)
+	}
+
+	// create target dir if necessary
+	err = provisioning.EnsureDir(fPath)
+	if err != nil {
+		log.Fatalf("failed to create target directory at %s", fPath)
+	}
+
+	// create manifests dir if necessary
+	manifestsDir := filepath.Join(fPath, provisioning.ManifestsDirName)
+	err = provisioning.EnsureDir(manifestsDir)
+	if err != nil {
+		log.Fatalf("failed to create manifests directory at %s", manifestsDir)
+	}
+}
+
+// NewCreateRAMPolicyCmd provides the "attach-ram-policy" subcommand
+func NewAttachRAMPolicyCmd() *cobra.Command {
+	attachRAMPolicyCmd := &cobra.Command{
+		Use:              "attach-ram-policy-to-existing-user",
+		Short:            "Attach RAM Policy to existing user",
+		Run:              attachRAMPolicyCmd,
+		PersistentPreRun: initEnvForAttachRAMPolicyCmd,
+	}
+
+	attachRAMPolicyCmd.PersistentFlags().StringVar(&AttachRAMPolicyOpts.Name, "name", "", "User-define name for all created Alibaba Cloud resources (can be separate from the cluster's infra-id)")
+	attachRAMPolicyCmd.MarkPersistentFlagRequired("name")
+	attachRAMPolicyCmd.PersistentFlags().StringVar(&AttachRAMPolicyOpts.CredRequestDir, "credentials-requests-dir", "", "Directory containing files of CredentialsRequests to create RAM AK for (can be created by running 'oc adm release extract --credentials-requests --cloud=alibabacloud' against an OpenShift release image)")
+	attachRAMPolicyCmd.MarkPersistentFlagRequired("credentials-requests-dir")
+	attachRAMPolicyCmd.PersistentFlags().StringVar(&AttachRAMPolicyOpts.Region, "region", "", "Alibaba Cloud region endpoint only required for GovCloud")
+	attachRAMPolicyCmd.MarkPersistentFlagRequired("region")
```

Comment:
The Alyun SDK appears to handle an unset region (eg ""). Why is the field required?